src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -37,14 +38,31 @@ public FunctionInvocationMiddleware(RequestDelegate next)
             // downstream middleware and filters rely on this
             context.Items.Add(ScriptConstants.AzureFunctionsHostManagerKey, manager);
             SetRequestId(context.Request);
-            await _next(context);
+            if (_next != null)
+            {
+                await _next(context);
+            }
 
             IFunctionExecutionFeature functionExecution = context.Features.Get<IFunctionExecutionFeature>();
 
+            int nestedProxiesCount = GetNestedProxiesCount(context, functionExecution);
+
+            IActionResult result = null;
+
             if (functionExecution != null && !context.Response.HasStarted)
             {
-                IActionResult result = await GetResultAsync(context, functionExecution);
+                result = await GetResultAsync(context, functionExecution);
+
+                if (nestedProxiesCount > 0)
+                {
+                    // if Proxy, the rest of the pipleline will be processed bt Proxies in case there are response overrides and what not.
+                    SetProxyResult(context, nestedProxiesCount, result);
+                    return;
+                }
+            }
 
+            if (result != null && !context.Response.HasStarted)
```

Comment:
We should be able to move this block to the above `if` block. `result` should also never be `null`.